src/datachain/data_storage/metastore.py

```diff
@@ -484,6 +488,38 @@ def create_checkpoint(
     ) -> Checkpoint:
         """Creates new checkpoint"""
 
+    #
+    # Dataset Version Jobs (many-to-many)
+    #
+
+    @abstractmethod
+    def link_dataset_version_to_job(
+        self,
+        dataset_version_id: int,
+        job_id: str,
+        is_creator: bool = False,
+        conn=None,
+    ) -> None:
+        """Link dataset version to job (many-to-many)."""
+
+    @abstractmethod
+    def get_ancestor_job_ids(self, job_id: str, conn=None) -> list[str]:
```

Comment:
Will this be sorted always? A helper to get the latest ancestor job id would be helpful/